streams/src/main/java/org/apache/kafka/streams/processor/TaskId.java

```diff
@@ -80,6 +83,35 @@ public String toString() {
         return namedTopology != null ? namedTopology + "_" + topicGroupId + "_" + partition : topicGroupId + "_" + partition;
     }
 
+    /**
+     * @throws TaskIdFormatException if the taskIdStr is not a valid {@link TaskId}
+     */
+    public static TaskId parse(final String taskIdStr) {
```

Comment:
I noticed this has been (re)moved since 2.8 so I put it back with updates to handle named topologies, plus tests which it did not seem to have

Reply:
If it was removed (and release in 2.8), why add it back? Seems we don't need it?

Reply:
Well that would be a breaking change by removing a non-deprecated API, no? And in this case I actually believe we should _not_ deprecate it -- if `toString` is part of the public TaskId API (and it should be) then this `parse` method which does the reverse should be as well. As I discussed with some during the KIP-740 ~debacle~ debate, part of the public contract of TaskId is in its string representation since that is what ends up in logs, metrics, etc. So imo it does make sense to provide a String-to-TaskId API